onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -2297,7 +2297,61 @@ Applies to session load, initialization, etc. Default is 0.)pbdoc")
         ORT_UNUSED_PARAMETER(ort_values);
         ORT_THROW("External initializers are not supported in this build.");
 #endif
-      });
+      })
+      .def("add_external_initializers_from_files_in_memory", [](PySessionOptions* options, py::list& names, py::list& buffers, py::list& lengths) -> void {
+#if !defined(ORT_MINIMAL_BUILD) && !defined(DISABLE_EXTERNAL_INITIALIZERS)
+        const auto num = names.size();
+        ORT_ENFORCE(num == buffers.size() && num == lengths.size(),
+                    "add_external_initializers_from_files_in_memory: expecting 'names', 'buffers' and 'lengths' to have equal length");
+
+        InlinedVector<PathString> file_names;
+        InlinedVector<std::pair<char*, const size_t>> files_buffers;
+        file_names.reserve(num);
+        files_buffers.reserve(num);
+
+        for (size_t i = 0; i < num; ++i) {
+          // names[i] must be str
+          ORT_ENFORCE(py::isinstance<py::str>(names[i]),
+                      "add_external_initializers_from_files_in_memory: each entry in 'names' must be a str");
+          file_names.emplace_back(ToPathString(py::str(names[i])));
```

Comment:
Suggest to declate this argument as std::vector<std::string> and let pybind do the types matching as well as issuing the error message.
This way let's not waste lines on this.